src/engine/Account-Imap.h

```diff
@@ -0,0 +1,154 @@
+/********************************************************************\
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ *                                                                  *
+\********************************************************************/
+/** @addtogroup Engine
+    @{ */
+/** @addtogroup Account
+    Account Import mapping functions
+    @{ */
+/** @file Account-Imap.h
+    @brief Account Imap handling public routines
+    @author Copyright (C) 1997 Robin D. Clark
+    @author Copyright (C) 1997-2003 Linas Vepstas <linas@linas.org>
+    @author Copyright (C) 2016 Robert Fewell
+*/
+
+#ifndef ACCOUNT_IMAP_H
+#define ACCOUNT_IMAP_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "qof.h"
+#include "gnc-engine.h"
+#include "policy.h"
+
+typedef struct imap_info
+{
+    Account        *source_account;
+    Account        *map_account;
+    GList          *list;
+    char           *category_head;
+    char           *full_category;
+    char           *match_string;
+    char           *count;
+}GncImapInfo;
+
+/* Predefined KVP paths */
+static const char* IMAP_FRAME = "import-map";
+static const char* IMAP_FRAME_BAYES = "import-map-bayes";
+
+#ifdef __cplusplus
+class GncImportMatchMap
+{
+public:
+    // Constructor - Destructor
+    GncImportMatchMap(Account *acc);
+    ~GncImportMatchMap();
+
+    Account *find_account (const char* category, const char *key);
+    void add_account (const char *category, const char *key, Account *acc);
+    void delete_account (const char *category, const char *key);
+
+    Account *find_account_bayes (GList* tokens);
+    void add_account_bayes (GList* tokens, Account *acc);
+
+    Account *get_account();
+    QofBook *get_book();
+
+private:
+    Account *account = nullptr;
+    QofBook *book;
+};
```

Comment:
Use m_ to prefix member variables. See http://wiki.gnucash.org/wiki/CodingStandard#C.2B.2B_Format.